Flaky UT TestTerminationOrderingSidecarsInReverseOrder of pkg/kubelet: kuberuntime

### Which jobs are flaking?

https://prow.ppc64le-cloud.cis.ibm.net/job-history/gs/ppc64le-kubernetes/logs/periodic-kubernetes-unit-test-ppc64le

### Which tests are flaking?

`TestTerminationOrderingSidecarsInReverseOrder` of `pkg/kubelet/kuberuntime/kuberuntime_termination_order_test.go`

### Since when has it been flaking?

Have observed this recently.

### Testgrid link

https://testgrid.k8s.io/ibm-unit-tests-ppc64le#Periodic%20unit%20test%20suite%20on%20ppc64le

### Reason for failure (if possible)

```
{Failed;  === RUN   TestTerminationOrderingSidecarsInReverseOrder
    kuberuntime_termination_order_test.go:179: expected delay for container sc3 = 1, got 0
--- FAIL: TestTerminationOrderingSidecarsInReverseOrder (4.00s)
;}
```

### Anything else we need to know?

Reproducible on a local x86_64 machine:
```
[root@maxwells1 kubernetes]# /root/go/bin/stress ./kuberuntime.test -test.run TestTerminationOrderingSidecarsInReverseOrder
5s: 0 runs so far, 0 failures
10s: 4 runs so far, 0 failures
---
---
3m50s: 172 runs so far, 0 failures
3m55s: 176 runs so far, 0 failures
4m0s: 180 runs so far, 0 failures

/tmp/go-stress-20240223T025228-4285150077
--- FAIL: TestTerminationOrderingSidecarsInReverseOrder (4.00s)
    kuberuntime_termination_order_test.go:179: expected delay for container sc2 = 2, got 1
    kuberuntime_termination_order_test.go:179: expected delay for container sc3 = 1, got 0
FAIL


ERROR: exit status 1

4m5s: 184 runs so far, 1 failures (0.54%)
4m10s: 188 runs so far, 1 failures (0.53%)
4m15s: 192 runs so far, 1 failures (0.52%)
```

### Relevant SIG(s)

/sig node

Seeing this flaking again:
```
[root@raji-x86-workspace1 kubernetes]#  /root/go/bin/stress ./kuberuntime.test -test.run TestTerminationOrderingSidecarsInReverseOrder
5s: 0 runs so far, 0 failures, 16 active
10s: 16 runs so far, 0 failures, 16 active
15s: 32 runs so far, 0 failures, 16 active
20s: 48 runs so far, 0 failures, 16 active
25s: 64 runs so far, 0 failures, 16 active

/tmp/go-stress-20250324T035630-25193102
--- FAIL: TestTerminationOrderingSidecarsInReverseOrder (4.00s)
    kuberuntime_termination_order_test.go:179: expected delay for container sc3 = 1, got 0
FAIL


ERROR: exit status 1

30s: 80 runs so far, 1 failures (1.25%), 16 active
35s: 96 runs so far, 1 failures (1.04%), 16 active
40s: 112 runs so far, 1 failures (0.89%), 16 active
45s: 128 runs so far, 1 failures (0.78%), 16 active
50s: 144 runs so far, 1 failures (0.69%), 16 active
55s: 160 runs so far, 1 failures (0.62%), 16 active
1m0s: 176 runs so far, 1 failures (0.57%), 16 active
1m5s: 192 runs so far, 1 failures (0.52%), 16 active
1m10s: 200 runs so far, 1 failures (0.50%), 16 active
1m15s: 214 runs so far, 1 failures (0.47%), 16 active
1m20s: 228 runs so far, 1 failures (0.44%), 16 active
^C
[root@raji-x86-workspace1 kubernetes]# uname -m
x86_64
[root@raji-x86-workspace1 kubernetes]#
```

I was able to reproduce this issue. The following are my steps:
```
go install golang.org/x/tools/cmd/stress@latest
go test -c ./pkg/kubelet/kuberuntime -race -count=1 -run TestTerminationOrderingSidecarsInReverseOrder
./kuberuntime.test -test.run TestTerminationOrderingSidecarsInReverseOrder
/home/sfogliato/go/bin/stress ./kuberuntime.test -test.run=TestTerminationOrderingSidecarsInReverseOrder
```
The following are my results:
```
5s: 0 runs so far, 0 failures, 16 active
10s: 16 runs so far, 0 failures, 16 active
15s: 32 runs so far, 0 failures, 16 active
20s: 48 runs so far, 0 failures, 16 active
25s: 64 runs so far, 0 failures, 16 active

/tmp/go-stress-20250430T091404-3676958160
--- FAIL: TestTerminationOrderingSidecarsInReverseOrder (4.00s)
    kuberuntime_termination_order_test.go:179: expected delay for container sc1 = 3, got 2
FAIL


ERROR: exit status 1

30s: 81 runs so far, 1 failures (1.23%), 16 active
35s: 96 runs so far, 1 failures (1.04%), 16 active
40s: 112 runs so far, 1 failures (0.89%), 16 active
45s: 128 runs so far, 1 failures (0.78%), 16 active
50s: 144 runs so far, 1 failures (0.69%), 16 active
55s: 160 runs so far, 1 failures (0.62%), 16 active
1m0s: 176 runs so far, 1 failures (0.57%), 16 active
1m5s: 192 runs so far, 1 failures (0.52%), 16 active
1m10s: 208 runs so far, 1 failures (0.48%), 16 active
1m15s: 224 runs so far, 1 failures (0.45%), 16 active
```

--- a/pkg/kubelet/kuberuntime/kuberuntime_termination_order_test.go
+++ b/pkg/kubelet/kuberuntime/kuberuntime_termination_order_test.go
@@ -125,7 +125,7 @@ func TestTerminationOrderingSidecarsInReverseOrder(t *testing.T) {
 	waitAndExit := func(name string) {
 		delay := int64(to.waitForTurn(name, 30))
 		delays.Store(name, delay)
-		time.Sleep(1 * time.Second)
+		time.Sleep(1250 * time.Millisecond)
 		to.containerTerminated(name)
 		wg.Done()
 	}

